npm/pkg/controlplane/controllers/v2/podcontroller.go

```diff
@@ -476,14 +462,19 @@ func (c *PodController) syncAddAndUpdatePod(newPodObj *corev1.Pod) error {
 	// Otherwise it returns list of deleted PodIP from cached pod's labels and list of added PodIp from new pod's labels
 	addToIPSets, deleteFromIPSets := util.GetIPSetListCompareLabels(cachedNpmPod.Labels, newPodObj.Labels)
 
+	newPodMetadata := dataplane.NewPodMetadata(podKey, newPodObj.Status.PodIP, newPodObj.Spec.NodeName)
+	// todo: verify pulling nodename from newpod,
+	// if a pod is getting deleted, we do not have to cleanup policies, so it is okay to pass in wrong nodename
+	cachedPodMetadata := dataplane.NewPodMetadata(podKey, cachedNpmPod.PodIP, newPodMetadata.NodeName)
 	// Delete the pod from its label's ipset.
 	for _, podIPSetName := range deleteFromIPSets {
 		klog.Infof("Deleting pod %s from ipset %s", cachedNpmPod.PodIP, podIPSetName)
 
-		// todo: verify pulling nodename from newpod
-		cachedPodMetadata := dataplane.NewPodMetadata(podKey, cachedNpmPod.PodIP, newPodMetadata.NodeName)
-
-		if err = c.dp.RemoveFromSets([]*ipsets.IPSetMetadata{ipsets.NewIPSetMetadata(podIPSetName, ipsets.KeyLabelOfPod)}, cachedPodMetadata); err != nil {
+		toRemoveSet := ipsets.NewIPSetMetadata(podIPSetName, ipsets.KeyLabelOfPod)
+		if util.IsKeyLabelSetName(podIPSetName) {
```

Comment:
misleading function name - `IsKeyValueLabelSetName`

Reply:
See previous comment about this.